examples/robot_net/computer.cc

```diff
@@ -47,29 +48,36 @@ main(int argc, char **argv)
         // Enable networking on Windows
         OS::Net::WindowsNetworking net;
 
-        // make connection to robot on default port
+        // Make connection to robot on default port
         Net::Client client(robotIP);
+
+        // Run client on background thread, catching any exceptions for rethrowing
+        BackgroundException::enableCatching();
         client.runInBackground();
 
-        // read video stream from network
+        // Read video stream from network
         Video::NetSource video(client);
 
-        // transmit motor commands over network
-        Robots::TankNetSink motor(client);
+        // Transmit motor commands over network
+        Robots::TankNetSink tank(client);
 
-        // add joystick for controlling Tank
+        // Add joystick for controlling robot
         HID::Joystick joystick;
-        motor.addJoystick(joystick); // send joystick events to motor
+        tank.addJoystick(joystick);
 
-        // display video stream
+        // Display video stream on screen
         Video::Display display(video, { 1240, 500 });
+        while(display.isOpen()) {
+            // Rethrow any exceptions caught on background thread
+            BackgroundException::check();
 
-        // poll joystick and video stream repeatedly
-        do {
-            if (!joystick.update() && !display.update()) {
+            // Poll joystick and camera for updates
+            bool joystickUpdate = joystick.update();
+            bool displayUpdate = display.update();
+            if (!joystickUpdate && !displayUpdate) {
```

Comment:
Yeah, it also meant that the display often wasn't opening, so the loop would exit immediately and it would quit. I assumed it was some other networking bug for ages